### PR TITLE
Add missing types

### PR DESCRIPTION
**Pull request by andrejak**
__what__

We started using the types from our client library in our dashboard and in doing so identified a few things we can improve and other bits that need fixing. This addresses the gaps we found.

Please see the comments for links to documentation to compare with.

### CODE REVIEW

File: src/booking/OrderCancellations/OrderCancellationsTypes.ts [hunk 1]
```diff
@@ -14,6 +14,43 @@ export interface ListOrderCancellationsParams extends PaginationMeta {
   order_id?: string
```
[the remark this answers is not in the document]
Comment (igorp1) [reply src/booking/OrderCancellations/OrderCancellationsTypes.ts:14]: done

File: src/booking/Orders/OrdersTypes.ts [hunk 2]
```diff
@@ -277,7 +291,47 @@ export interface OrderSliceSegment {
...
-  distance?: string | null
+  distance: string
+
+  /**
+   * Additional segment-specific information about the stops, if any, included in the segment
+   */
+  stops: Array<Stop>
+}
+
+export interface Stop {
```
Comment (andrejak) [inline src/booking/Orders/OrdersTypes.ts:302]: https://duffel.com/docs/api/v2/orders#orders-schema-slices-slices-segments-slices-segments-stops

File: src/booking/Orders/OrdersTypes.ts [hunk 3]
```diff
@@ -365,6 +424,12 @@ export interface OrderDocument {
    * The type of document
    */
   type: OrderDocumentsType
+
+  /**
+   * The list of passenger ids the document applies to
+   * @example ["pas_00009hj8USM7Ncg31cBCLL"]
+   */
+  passenger_ids: string[]
```
Comment (andrejak) [inline src/booking/Orders/OrdersTypes.ts:432]: https://duffel.com/docs/api/v2/orders#orders-schema-documents-documents-passenger-ids

File: src/booking/Orders/OrdersTypes.ts [hunk 4]
```diff
@@ -520,6 +585,15 @@ export interface Order {
...
   available_actions: OrderAvailableAction[]
+
+  /**
+   * Airlines are always the source of truth for orders. The orders returned in the Duffel API
+   * are a view of those orders. This field is the ISO 8601 datetime at which the order was
+   * last synced with the airline. If this datetime is in the last minute you can consider the order up to date.
+   *
+   * @example "2020-04-11T15:48:11Z"
+   */
+  synced_at: string
```
Comment (andrejak) [inline src/booking/Orders/OrdersTypes.ts:596]: https://duffel.com/docs/api/v2/orders#orders-schema-synced-at

File: src/booking/AirlineInitiatedChanges/AirlineInitiatedChangesTypes.ts [hunk 5]
```diff
@@ -11,6 +11,13 @@ export type AirlineInitiatedChangeAvailableAction =
   | 'change'
   | 'update'
 
+export interface TravelAgentTicket {
```
Comment (igorp1) [inline src/booking/AirlineInitiatedChanges/AirlineInitiatedChangesTypes.ts:14]: Should we add comment on what this is?

```suggestion
/**
 * The associated Travel Agent Ticket, if any, for this Airline-Initiated Change.  
 * This value will be present for Airline-Initiated changes that take some time to be processed.
 */
export interface TravelAgentTicket {
```

File: src/types/shared.ts [hunk 6]
```diff
@@ -112,28 +147,7 @@ export type FlightsConditions = {
...
-     */
-    penalty_amount?: string | null
-    /**
-     * Whether this kind of modification is allowed post-booking
-     *
-     * @example "true"
-     */
-    allowed: boolean
-  } | null
+  change_before_departure?: FlightsCondition | null
```
Comment (igorp1) [inline src/types/shared.ts:150]: I don't think we should allow `undefined` in the types, they might mismatch at times if the only expectation is `Something` or `null`.
  
```suggestion
  change_before_departure: FlightsCondition | null
```

File: src/types/shared.ts [hunk 7]
```diff
@@ -81,28 +137,7 @@ export type FlightsConditions = {
...
-     */
-    penalty_amount?: string | null
-    /**
-     * Whether this kind of modification is allowed post-booking
-     *
-     * @example "true"
-     */
-    allowed: boolean
-  } | null
+  refund_before_departure?: FlightsCondition | null
```
Comment (igorp1) [inline src/types/shared.ts:140]:
```suggestion
  refund_before_departure: FlightsCondition | null
```

File: src/types/shared.ts [hunk 8]
```diff
@@ -148,6 +162,7 @@ export type FlightsConditions = {
  * passengers if you require this level of granularity.
  */
 export type OfferSliceConditions = FlightsConditions & {
+  // TODO: ask James about this
```
Comment (igorp1) [inline src/types/shared.ts:165]: I left this comment here yesterday but after reading it again, I've answered my own question (I thought refund conditions would not be here, but they are, so all good). Safe to remove!

 ```suggestion
```

**Review by igorp1 — APPROVED** [review]
Thank you for all the guiding comments to the docs, very helpful to quickly validate the change (and possibly for archeology in the future!)

File: src/booking/Offers/OfferTypes.ts [hunk 9]
```diff
@@ -450,7 +451,7 @@ export interface OfferSlice {
    *   - `"4"`: premium seating, additional legroom and recline. Situated in business class or higher.
    *    - `"5"`: deluxe seating, additional legroom and reclines to lie flat position. Situated in business class or higher.
    */
-  ngs_shelf: number
+  ngs_shelf: number | null
```
Comment (andrejak) [inline src/booking/Offers/OfferTypes.ts:454]: https://duffel.com/docs/api/v2/offers#offers-schema-slices-slices-ngs-shelf

File: src/types/shared.ts [hunk 10]
```diff
@@ -151,17 +165,17 @@ export type OfferSliceConditions = FlightsConditions & {
   /**
    * Whether passengers are able to select a seat prior to check in.
    */
-  advance_seat_selection: boolean
+  advance_seat_selection: boolean | null
```
Comment (andrejak) [inline src/types/shared.ts:168]: https://duffel.com/docs/api/v2/offers#offers-schema-slices-slices-conditions-slices-conditions-advance-seat-selection

File: src/types/shared.ts [hunk 11]
```diff
@@ -162,21 +162,20 @@ export type FlightsConditions = {
...
    * Whether passengers are given preferential boarding over others passengers in their cabin.
    */
-  priority_boarding: boolean
+  priority_boarding: boolean | null
 
   /**
    * Whether passengers are given access to a fast track lane during check in.
    */
-  priority_check_in: boolean
+  priority_check_in: boolean | null
```
Comment (jekku) [inline src/types/shared.ts:178]: https://duffel.com/docs/api/v2/offers#offers-schema-slices-slices-conditions-slices-conditions-priority-check-in ✅

File: src/types/shared.ts [hunk 12]
```diff
@@ -162,21 +162,20 @@ export type FlightsConditions = {
...
    * Whether passengers are able to select a seat prior to check in.
    */
-  advance_seat_selection: boolean
+  advance_seat_selection: boolean | null
 
   /**
    * Whether passengers are given preferential boarding over others passengers in their cabin.
    */
-  priority_boarding: boolean
+  priority_boarding: boolean | null
```
Comment (jekku) [inline src/types/shared.ts:173]: https://duffel.com/docs/api/v2/offers#offers-schema-slices-slices-conditions-slices-conditions-priority-boarding ✅

File: src/booking/Orders/mockOrders.ts [hunk 13]
```diff
@@ -597,7 +599,10 @@ export const mockOnHoldOrders: Order[] = [
             iata_code: 'MIA',
           },
         },
-        conditions: { change_before_departure: null },
+        conditions: {
```
Comment (jekku) [inline src/booking/Orders/mockOrders.ts:602]: https://duffel.com/docs/api/v2/orders/schema#orders-schema-conditions ✅

**Review by jekku — APPROVED** [review]

**Pull request merged**